ts/features/ingress/screens/IngressScreen.tsx

```diff
@@ -0,0 +1,135 @@
+import { isBackendStatusLoadedSelector } from "../../../store/reducers/backendStatus";
+import { setIsBlockingScreen } from "../store/actions";
+import ModalSectionStatusComponent from "../../../components/SectionStatus/modal";
+import { isMixpanelInitializedSelector } from "../../mixpanel/store/selectors";
+import {
+  trackIngressCdnSystemError,
+  trackIngressServicesSlowDown,
+  trackIngressTimeout
+} from "../analytics";
+
+const TIMEOUT_CHANGE_LABEL = (5 * 1000) as Millisecond;
+const TIMEOUT_BLOCKING_SCREEN = (10 * 1000) as Millisecond;
+
+export const IngressScreen = () => {
+  const isMixpanelInitialized = useIOSelector(isMixpanelInitializedSelector);
+  const isMixpanelEnabled = useIOSelector(isMixpanelEnabledSelector);
+  const dispatch = useIODispatch();
+  const [netInfo, setNetInfo] = useState<NetInfoState>();
+  const [showBlockingScreen, setShowBlockingScreen] = useState(false);
+  const [contentTitle, setContentTitle] = useState(I18n.t("startup.title"));
+
+  useEffect(() => {
+    // `isMixpanelEnabled` mustn't be `false`
+    if (isMixpanelInitialized && isMixpanelEnabled !== false) {
+      trackIngressScreen();
+    }
+  }, [isMixpanelInitialized, isMixpanelEnabled]);
+
+  useEffect(() => {
+    const timeouts: Array<number> = [];
+    timeouts.push(
+      setTimeout(() => {
+        setContentTitle(I18n.t("startup.title2"));
+        timeouts.shift();
+      }, TIMEOUT_CHANGE_LABEL)
+    );
+    timeouts.push(
+      setTimeout(() => {
+        setShowBlockingScreen(true);
+        dispatch(setIsBlockingScreen());
+        timeouts.shift();
+      }, TIMEOUT_BLOCKING_SCREEN)
+    );
+    void fetchNetInfo()
+      .then(info => {
+        if (!info.isConnected) {
+          timeouts.forEach(clearTimeout);
+        }
+        setNetInfo(info);
+      })
+      .catch();
+
+    return () => {
+      timeouts?.forEach(clearTimeout);
+    };
+  }, [dispatch]);
+
+  if (netInfo && !netInfo.isConnected) {
+    return (
+      <OperationResultScreenContent
+        testID="device-connection-lost-id"
+        pictogram="lostConnection"
+        title={I18n.t("startup.connection_lost.title")}
+        subtitle={I18n.t("startup.connection_lost.description")}
+      />
+    );
+  }
+
+  if (showBlockingScreen) {
+    return <IngressScreenBlockingError />;
+  }
+
+  return (
+    <>
+      <ModalSectionStatusComponent
+        sectionKey="ingress"
+        sticky
+        trackingAction={trackIngressServicesSlowDown}
+      />
+      <LoadingScreenContent
+        testID="ingress-screen-loader-id"
+        contentTitle={contentTitle}
+      />
+    </>
+  );
+};
+
+const IngressScreenBlockingError = memo(() => {
+  const isBackendStatusLoaded = useIOSelector(isBackendStatusLoadedSelector);
+  const isMixpanelEnabled = useIOSelector(isMixpanelEnabledSelector);
+
+  useEffect(() => {
+    // It's not necessary to check if mixpanel is initialized since this screen is shown after 10 seconds.
```

Comment:
```suggestion
    // It's not necessary to check if mixpanel is initialized since this screen is shown after 10 seconds.
    // If mixpanel is not initialized at that time, we have an issue spanning, system-wide.
```